install/troubleshooting/collector_hang.mdx

```diff
@@ -0,0 +1,48 @@
+---
+title: Collector hangs
+backlink_href: /docs/install/troubleshooting
+backlink_title: 'Installation Troubleshooting'
+---
+
+If you find the collector hanging either periodically or constantly, or even within the test run,
+the most likely cause is `pg_stat_statements` statistic data being big and taking long time to query the statistic from it.
+This especially happens when there are many queries with long query texts.
+
+You can check out below points using the following query to identify if ps_stat_statements statistic data is big or not.
+
+* How many queries does `pg_stat_statements` statistic data currently have
+* The maximum number of queries `pg_stat_statements` keeps track of (default: 5000)
+* Total size of tracked queries
+
+```sql
+SELECT
+  count(*),
+  current_setting('pg_stat_statements.max'),
+  pg_size_pretty(sum(length(query)))
+FROM pg_stat_statements;
+```
+
+To mitigate the issue, you can run `SELECT pg_stat_statements_reset()` to reset the statistic data.
+When the collector test run (`pganalyze-collector --test --verbose`) is already hanging, it's good to try this out and see the test will stop hanging.
+
+If the collector still hangs after this, please [contact us](/contact) so that we can help debugging further.
+
+Once you confirm that `pg_stat_statements_reset()` is a valid mitigation, you can enable a recurring reset executed by pganalyze.
+
+## Setting up recurring `pg_stat_statements_reset()`
+
+First, you need to create the following helper method:
+
+```sql
+CREATE OR REPLACE FUNCTION pganalyze.reset_stat_statements() RETURNS SETOF void AS
+$$
+  /* pganalyze-collector */ SELECT * FROM public.pg_stat_statements_reset();
+$$ LANGUAGE sql VOLATILE SECURITY DEFINER;
+```
+
+Second, go to the Server Settings panel in Settings page and tune "Collector: Query Statistics Reset Frequency":
+
+![Server Settings](/server_settings.png)
+
+Please feel free to [reach out to us](/contact) for which frequency to pick.
+You could always start from 24 hours to see if that helps the situation.
```

Comment:
I think if people do reach out, we'll probably just tell them to start at 24h and make it more frequent if necessary, so might as well start with that here:
```suggestion
Start with a reset frequency of 24 hours to see if that helps the situation. If this helps for a while but the problem eventually happens again, try increasing the frequency. Please feel free to [reach out to us](/contact) to tune it further.
```